tests/server/conftest.py

```diff
@@ -120,18 +120,16 @@ def inner(
 
         response_ids = [struct["id"] for struct in response["data"]]
 
-        if expected_return is None:
-            expected_return = len(expected_ids)
+        if expected_return is not None:
+            assert expected_return == response["meta"]["data_returned"]
```

Comment:
Does this change now that `data_returned` is actually the number of results for the query, not the number of results left unpaginated?


If not, then this is ready to be merged.

Reply:
Looking at this again, I think the single structure testing needs to distinguish between a filter that returns one result, and a request to a single entry endpoint.

I have a commit that does the following:

- If `expected_ids = ["mpf_1"]`, validate as a single response to a multi-entry query.
- If `expected_ids = "mpf_1"`, validate as a single-entry query.

Reply:
We use len(expected_ids) in the inner function of check_response, so it should not be a string, as in that case it would return the number of characters instead of the number of ids.

Reply:
My commit changes it to `expected_ids = [expected_ids]` if it finds a string, so I think we're good. I also tweaked your test for it and it works fine.

Reply:
There is some lag. When I wrote my comment I had not seen your commit yet.